olive/passes/onnx/quantization.py

```diff
@@ -636,16 +685,46 @@ def _run_for_config(
         from onnxruntime import __version__ as OrtVersion
 
         if version.parse(OrtVersion) < version.parse("1.16.2"):
-            raise OlivePassError("MatMul4BitsQuantizer is only supported in onnxruntime >= 1.16.2")
+            raise ValueError("MatMul4BitsQuantizer is only supported in onnxruntime >= 1.16.2")
 
         from onnxruntime.quantization.matmul_4bits_quantizer import MatMul4BitsQuantizer
 
         output_model_path = resolve_onnx_path(output_model_path, Path(model.model_path).name)
 
+        weight_only_quant_config_class = None
+        weight_only_quant_config = None
+        algo_config = deepcopy(config["weight_only_quant_configs"] or {})
         if version.parse(OrtVersion) >= version.parse("1.17.0"):
-            from onnxruntime.quantization.matmul_4bits_quantizer import WeightOnlyQuantConfig
+            from onnxruntime.quantization.matmul_4bits_quantizer import (
+                GPTQWeightOnlyQuantConfig,
+                RTNWeightOnlyQuantConfig,
+            )
+
+            if config["algorithm"] == "RTN":
+                weight_only_quant_config_class = RTNWeightOnlyQuantConfig
+            elif config["algorithm"] == "GPTQ":
+                if "block_size" in algo_config and version.parse(OrtVersion) < version.parse("1.18.0"):
+                    # ort 1.17.0+ uses blocksize instead of block_size :(
+                    algo_config["blocksize"] = algo_config["block_size"]
+                    algo_config.pop("block_size")
+                dataloader = get_calibration_dataloader(data_root, self._user_module_loader, config)
+                weight_only_quant_config_class = partial(GPTQWeightOnlyQuantConfig, calibration_data_reader=dataloader)
+
+            if version.parse(OrtVersion) >= version.parse("1.18.0"):
+                from onnxruntime.quantization.matmul_4bits_quantizer import (
+                    DefaultWeightOnlyQuantConfig,
+                    HQQWeightOnlyQuantConfig,
+                )
+
+                if config["algorithm"] == "DEFAULT":
+                    weight_only_quant_config_class = DefaultWeightOnlyQuantConfig
+                elif config["algorithm"] == "HQQ":
+                    weight_only_quant_config_class = HQQWeightOnlyQuantConfig
+            if version.parse(OrtVersion) < version.parse("1.18.0") and config["algorithm"] in ("HQQ", "DEFAULT"):
```

Comment:
I think this is equivalent to `elif config["algorithm"] in ....`?

Reply:
Tue and actually it is suggested by copilot as well.

At that time, I just want to make it clear then change it. :)

Anyway, I will change back and add comment to make it clear.

Reply:
Thanks. To me, the elif condition looks more clear. Otherwise, it seems like it is an unrelated condition. But it is nit

Reply:
Updated.